tutorials/hello/hello_client/src/app/plugins/hello_info/endpoint.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable prefer-const */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable dot-notation */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/await-thenable */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable import/no-cycle */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable no-console */
+/* eslint-disable spaced-comment */
+import {
+    BasePluginEndpoint,
+    PluginEndpointContext,
+    BasePlugin,
+    db as liskDB,
+    cryptography,
+    codec,
+} from 'lisk-sdk';
+import {
+    getEventHelloInfo,
+    getLastCounter,
+    getLastEventHeight,
+    setEventHelloInfo,
+    setLastCounter,
+    setLastEventHeight
+} from './db';
+import { chainEventSchema } from './schemas';
+
+
+export class Endpoint extends BasePluginEndpoint {
+    private _client!: BasePlugin['apiClient'];
+    private _db!: liskDB.Database;
+
+    public init(db: liskDB.Database, apiClient: BasePlugin['apiClient']) {
+        this._db = db;
+        this._client = apiClient;
+    }
+
+    // Returns all Sender Addresses, Hello Messages and Block Height of the block where the Hello Event was emitted.
+    public async getAddressList(_context: PluginEndpointContext): Promise<any[]> {
+        let addressList: any;
+        const data = [];
+        const lastCounter = await getLastCounter(this._db);
+        for (let index = 1; index <= lastCounter.counter; index += 1) {
+            addressList = await getEventHelloInfo(this._db, index);
+            data.push({
+                ID: index,
+                Sender_Address: cryptography.address.getLisk32AddressFromAddress(addressList.senderAddress),
+                Message: addressList.message,
+                Block_Height: addressList.height,
+            })
+        }
+        return data;
+    }
+
+    // Syncs on-chain event's data with plugin's database.
+    public async syncChainEvents() {
```

Comment:
Would be nice if the events are updated in a specific interval automatically. The size of the interval could be configured in the config, by providing a new config option, to set the sync interval (counted in blocks)

Reply:
The old config usecase has been updated.

I have added a new `syncInterval` config property, allowing my plugin to sync after an interval. The plugin will now store any newly generated **HelloEvents** in the plugin's database by checking after a specific interval.